apps/src/studio/studio.js

```diff
@@ -521,6 +476,73 @@ function overlappingTest(x1, x2, xVariance, y1, y2, yVariance) {
   return (Math.abs(x1 - x2) < xVariance) && (Math.abs(y1 - y2) < yVariance);
 }
 
+Studio.allGoals_ = function () {
+  return Studio.spriteGoals_.concat(Studio.dynamicSpriteGoals_);
+};
+
+Studio.drawGoal = function (i, goal) {
```

Comment:
I would name this `createGoal` since it is really creating all of the elements. I think of a "draw" function as something that would happen frequently.